### PR TITLE
[CARBONDATA-1284]Implement hive based schema storage in carbon

### PR DESCRIPTION
**Pull request by ravipesala**
Implemented hive based schema storage in carbon
By using the property `spark.carbon.hive.schema.store` to `true` , carbon can only save schema to hive metastore. it will not store in file system.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/373/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2961/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/374/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2962/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/375/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2963/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/391/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2980/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2987/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/398/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/409/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2998/

File: integration/spark2/src/main/scala/org/apache/carbondata/spark/util/CarbonSparkUtil.scala [hunk 1]
```diff
@@ -42,4 +44,14 @@ object CarbonSparkUtil {
       }
     CarbonMetaData(dimensionsAttr, measureAttr, carbonTable, DictionaryMap(dictionary.toMap))
   }
+
+  def createCarbonRelation(tableInfo: TableInfo, tablePath: String): CarbonRelation = {
+    val identifier = AbsoluteTableIdentifier fromTablePath(tablePath)
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/util/CarbonSparkUtil.scala]: AbsoluteTableIdentifier.fromTablePath(tablePath)

Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/carbondata/spark/util/CarbonSparkUtil.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3019/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/431/

**Comment by jackylk** [comment]
LGTM
Test case need to be solved in another PR

**Comment by gvramana** [comment]
@ravipesala Previously schema refresh is using store level timestamp file and then table schema file modification  timestamp. How now refresh is detected?

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/549/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3141/

**Pull request closed without merging**